compiler/src/main/kotlin/io/github/apl_cornell/viaduct/codegeneration/BackendCodeGenerator.kt

```diff
@@ -91,6 +98,20 @@ private class BackendCodeGenerator(
     /** Generates code for [host]'s role in the function [functionDeclaration]. */
     private fun generate(functionDeclaration: FunctionDeclarationNode): FunSpec {
         val hostFunctionBuilder = FunSpec.builder(functionDeclaration.name.value.name)
```

Comment:
You should mark all functions except `main` as private. You can determine `main` using `Declarations.mainFunction`.